### PR TITLE
[doc] [ltac2] Build Ltac2 documentation

### LINKED ISSUE #10139: Ltac2 coqdoc-generated documentation is not built / installed.
@ppedrot wants users to refer to this documentation frequently instead of crowding the manual because many things are defined there instead of being primitive. But at the current time, it is not installed: https://coq.github.io/doc/master/stdlib/. This is consistent with the name and URL of this documentation BTW (if we consider that stdlib is the name of everything under the `Coq` prefix). This needs some thinking IMO, and is not so urgent, as long as it is done in time for 8.11.

### PR DESCRIPTION
**Pull request by ejgallego**
This fixes #10139 ; we now build the Ltac2 documentation and
deploy it.

**Kind:** documentation / bug fix

### CODE REVIEW

File: Makefile.doc [hunk 1]
```diff
@@ -178,12 +180,12 @@ doc/stdlib/FullLibrary.tex: doc/stdlib/Library.tex
 ifdef QUICK
 doc/stdlib/FullLibrary.coqdoc.tex:
 	$(COQDOC) -q -boot --gallina --body-only --latex --stdout --utf8 \
-            -R theories Coq -R plugins Coq $(VFILES) > $@
+            $(DOCLIBS) Coq $(VFILES) > $@
```
Comment (Zimmi48) [inline Makefile.doc]: Leftover here:

```suggestion
            $(DOCLIBS) $(VFILES) > $@
```

**Review by Zimmi48 — APPROVED** [review]
That's great, thank you!
It looks like there was a typo in your changes (see below) but testing shows that this doesn't prevent the build nor hinders the result.
I'm a bit puzzled by the (seemingly random) ordering of the current stdlib doc index, but your patch is not making things worse, although it could make sense to reorder everything alphabetically, or to have a clear split with section titles between the stdlib, and the plugins, and possibly between the Coq namespace, and the rest.

File: Makefile.doc (hunk 1 above)
Comment (ejgallego) [reply Makefile.doc]: Ups, will fix.

**Comment by ejgallego** [comment]
> I'm a bit puzzled by the (seemingly random) ordering of the current stdlib doc index, but your patch is not making things worse, although it could make sense to reorder everything alphabetically, or to have a clear split with section titles between the stdlib, and the plugins, and possibly between the Coq namespace, and the rest.

Indeed, I placed Ltac2 next to ssreflect as a kind of clustering; I'd say we need to go quite further beyond what you propose in terms of library documentation tho; I had some chat already with @herbelin and took some notes, but very briefly IMHO we should try to have something such `hackage` directly.

**Comment by Zimmi48** [comment]
I don't understand your comment. I was speaking of a low-cost short-term improvement in terms of organizing the index of the stdlib doc, not about the more general problem of giving documentation to all packages available in the index (which is indeed a related but broader topic, and which will take longer to fix).

**Review by Zimmi48 — APPROVED** [review]
This is good to go in any case.

**Comment by ejgallego** [comment]
> I don't understand your comment. I was speaking of a low-cost short-term improvement in terms of organizing the index of the stdlib doc, not about the more general problem of giving documentation to all packages available in the index (which is indeed a related but broader topic, and which will take longer to fix).

We could indeed tweak a bit the index, but what I mean is that even if included in the Coq repos, ltac2 behaves closer to an independent package; it wouldn't surprise me if Coq 8.13 for example should feature in the "index" mtac2, equations, coqsmt / hammer and metacoq. Thus I think it cannot hurt to start thinking how the "index" should be organized; maybe in terms of categories of packages.

**Comment by Zimmi48** [comment]
Well, yes. And this is not unrelated to the platform effort in fact. Let's talk some more about it when we get the chance to meet face to face.

**Pull request merged**